File: project-euler/Python/problem1.py
#!/usr/bin/env python
"""
Python solution for Project Euler Problem 1: Multiples of 3 and 5
"""
__author__ = "Emelie Widegren"
__email__ = "emeliewide@gmail.com"


def multiples():
    """
    Returns the sum of all the multiples of 3 or 5 below 1000.
    """
    result = 0
    for i in range(1000):
        if (i % 3) == 0 or (i % 5) == 0:
            result += i
    return result


if __name__ == '__main__':
    print(multiples())

File: project-euler/Java/Problem1.java
/**
 * Java solution for Project Euler Problem 1: Multiples of 3 and 5
 *
 * @author Emelie Widegren <widegren.emelie@gmail.com>
 */
public class Problem1 {

  public static void main(String arg[]) {
    System.out.println(multiples());
  }

  /** Returns the sum of all the multiples of 3 or 5 below 1000. */
  private static int multiples() {
    int result = 0;
    for (int i = 0; i < 1000; i++) {
      if (i % 3 == 0 || i % 5 == 0) {
        result += i;
      }
    }
    return result;
  }
}
